File: data/autospec_bench/code2inv_133_benchmark_verified/7_verified.c
#include<limits.h>

int unknown();

/*@
  requires 0 <= x <= 10;
  requires 0 <= y <= 10;
*/
void foo(int x, int y) {
  /*@
  loop invariant x == 20 ==> y != 0;
  loop invariant 0 <= y;
  loop invariant 0 <= x;
  loop assigns y;
  loop assigns x;
  */
  while (unknown()) {
    if (x > 2147483647 - 10 || y > 2147483647 - 10){
      break;
    }
    x  = x + 10;
    y  = y + 10;
  }
  if (x == 20) {
    //@ assert y != 0;
  }
}

File: RQ1_result/migration_success/7_verified.rs
#[allow(unused_imports)]
use vstd::math::*;
use vstd::prelude::*;
use vstd::slice::*;
verus! {

#[verifier::external_body]
fn unknown() -> (result: i32) {
    unimplemented!();
}

#[verifier::loop_isolation(false)]
#[verifier::exec_allows_no_decreases_clause]
fn foo(mut x: i32, mut y: i32)
    requires
        0 <= x <= 10,
        0 <= y <= 10,
{
    while unknown() != 0
        invariant
            (x == 20) as int != 0 ==> (y != 0) as int != 0,
            0 <= y,
            0 <= x,
    {
        if x > i32::MAX - 10 || y > i32::MAX - 10 {
            break ;
        }
        x = x + 10;
        y = y + 10;
    }
    if x == 20 {
        assert(y != 0);
    }
}

fn main() {
}

} // verus!
